### ISSUE TITLE
Excessive Media Requests Delay Critical Post Operations

### ISSUE THREAD

**Issue by fluiddot** [wordpress-mobile/WordPress-Android#20905]
### Expected behavior
- The app only generates one request to the media endpoint for each featured image of the posts to be fetched.
- Post management operations aren't locked due to fetching featured images of posts.

### Actual behavior
When opening the app for the first time and navigating a site with a long list of posts with featured images, the app generates significantly more requests to the media endpoint than the number of posts. For instance, 40 posts with featured images resulted in 177 requests. This issue is exacerbated by slow network connections, leading to even more requests.

During this period, critical post operations like saving, publishing, and deleting are locked until all the featured image requests are completed. This results in a situation where post management is blocked for several minutes, severely impacting the user experience.

### Steps to reproduce the behavior

**Preparation:** 
- Create multiple posts (20 or more) and attach a featured image. This can be done quickly by using the [WPCOM console](https://developer.wordpress.com/docs/api/console/) and the endpoint rest/v1.1/sites/${siteId}/posts/new.

#### Extra requests to the media endpoint
1. On the device/emulator, clear storage of the app, or delete and reinstall the app.
2. Open the app and navigate to the site.
3. Connect the device/emulator to the Charles proxy.
4. Open the app and go to the Post screen.
5. Observe all extra requests to the media endpoint in the Charles proxy.

#### Post manage operations locked
1. On the device/emulator, clear storage of the app, or delete and reinstall the app.
2. Open the app and navigate to the site.
3. Connect the device/emulator to the Charles proxy.
4. Slow down the network connection. This can be achieved by enabling throttling in the Charles proxy.
5. Open the app and go to the Post screen.
6. Edit a post and save it.
7. Observe that the saving operation takes a significant delay.

**Comment by dangermattic** [comment]
Thanks for reporting! 👍

**Comment by fluiddot** [comment]
After investigating the issue, I noticed the following:

- The request to the media endpoint is related to the featured image of posts shown in the post list ([reference](https://github.com/wordpress-mobile/WordPress-Android/blob/f8db3db3ee1f2ce13c8da6344aaa2353a2bd6131/WordPress/src/main/java/org/wordpress/android/ui/posts/PostListFeaturedImageTracker.kt#L27-L47)).
- The function that fetches the featured image (`getFeaturedImageUrl`) dispatches a request if the media item is not cached. However, it doesn't check if it was requested before. This results in extra requests for the same item as outlined in the issue.
- Going up in the stack trace, I see that all calls are triggered by `PagedListPositionalDataSource.loadInitial` ([reference](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/7d5b075778e2c2eb7195e296ed6b8c3df8817356/fluxc/src/main/java/org/wordpress/android/fluxc/model/list/PagedListFactory.kt#L40)). However, I saw that this function is called multiple times with the same data set. I don't have the context to understand why this is happening, but I presumed that it has to do with the paging logic of the Android library `androidx.paging`.

_Stack trace of a call to `getFeaturedImageUrl`:_
<img width="1299" alt="Screenshot 2024-05-29 at 12 00 47" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14905380/9c33263c-56d7-4605-9c48-aeb5b384a25e">
